### PR TITLE
Downgrade Swift version to 5.0

### PR DESCRIPTION
**Pull request by slashmo**
Adds support for Swift 5.0 & Swift 5.1 which is necessary for being integrated by other libraries that still support these versions. The CI now executes tests in parallel on `5.0.3`, `5.1`, and `5.2`.

### CODE REVIEW

File: Sources/Baggage/BaggageContext.swift [hunk 1]
```diff
@@ -70,14 +70,14 @@ public struct BaggageContext: BaggageContextProtocol {
...
-            self.value as! Key.Value
+            return self.value as! Key.Value
         }
     }
 }
 
 extension BaggageContext: CustomStringConvertible {
     public var description: String {
-        "\(Self.self)(keys: \(self._storage.map(\.key.name)))"
+        return "\(type(of: self).self)(keys: \(self._storage.map { $0.key.name }))"
```
Comment (slashmo) [inline Sources/Baggage/BaggageContext.swift:80]: 😭

Comment (ktoso) [reply Sources/Baggage/BaggageContext.swift:80]: lol oh well, not too bad 😉

**Review by ktoso — APPROVED** [review]
LGTM, thanks for taking this on

**Comment by ktoso** [comment]
I cleaned up milestones a bit, this will be 0.3.0

**Pull request merged**